### PR TITLE
Fix RTL text direction in PDF export

### PR DESCRIPTION
**Pull request by seluianova**
### Short description
Somehow RTL languages are printed LTR in the PDF export

Title has correct direction, but content is incorrect:
image

### Proposed changes
Text direction is broken when text passes through pdf_strip_fontstyles filter.
Adding "unescape" in return statement solves the issue.

### Side effects
I also updated pdfs in tests for other languages, because tests were failing.
But visually there was no difference except for Arabisch.
pdf_test_result

### Resolved issues
Fixes: #1954

__________________________________________________

[Pull Request Review Guidelines](https://digitalfabrik.github.io/integreat-cms/pull-request-review-guide.html)

### CODE REVIEW

**Review by JoeyStk — CHANGES_REQUESTED** [review]
Thank you so much for this PR. It took me a while to wrap my head around the issue. For Arabic this works for me well now. However for Farsi I still have the issue. 
To me, the part that confused me the most, is that not only the direction of the entire line is mixed up, the direction is also mixed up inside a word. Since in Arabic and Farsi every letter looks different dependent on the position it has within a word, the words don't even look the same in Pdf and the editor.
grafik

**Comment by seluianova** [comment]
@JoeyStk 
Cool finding, thanks. I was looking at Arabisch for so long that my eyes blurred.

I have just reviewed Arabic and Farsi one more time, and here is my summary:
As I see, issues below are actually the same for both Arabic and Farsi.

1. When Arabic/Farsi sentence **contains a Latin word**, weird metamorphoses take place:
I've marked identical pieces of text with the same color. So, we can see that these pieces of text change their position relative to the English word.
What is besonders strange to me - they change not their direction (they still RTL), but position in the sentence.

2. When there is a line break, the newline goes up, not down.
3. Bullet points are on the left, but should be on the right.

image

**Comment by svenseeberg** [comment]
Have you tried setting an `dir="rtl"` attribute for a list item, for example? Would that improve the layout?

**Comment by seluianova** [comment]
As we discussed on the team call:
- the first issue related to mixed content - we will leave it as is
- for second (line breaks) and third (bullet points) and ToC - I will raise issues for xhtml2pdf library

**Review by JoeyStk — APPROVED** [review]
For the new scope we're going for, everything seems to work.
I have tested it again and didn't run into issues :)

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,6 +5,7 @@ UNRELEASED
 * [ [#1345](https://github.com/digitalfabrik/integreat-cms/issues/1345) ] Track DeepL API usage by regions
 * [ [#1695](https://github.com/digitalfabrik/integreat-cms/issues/1695) ] Treat URLs with broken hash anchors as valid in link checker
 * [ [#1969](https://github.com/digitalfabrik/integreat-cms/issues/1969) ] Hide edit link button in content forms for users without edit permission
+* [ [#1954](https://github.com/digitalfabrik/integreat-cms/issues/1954) ] Fix RTL text direction in PDF export
```
Comment (timoludwig) [inline CHANGELOG.md:8]: Due to the long time this PR is open now, this needs to be rebased and then moved up into the unreleased section again.

**Review by timoludwig — CHANGES_REQUESTED** [review]
Thanks a lot! :+1:
I hope that with this fix, the Arabic texts are at least partly readable with some effort...

**Review by timoludwig — APPROVED** [review]

**Pull request merged**